docs/merkletree/treetex/main.go

```diff
@@ -197,31 +202,32 @@ func openInnerNode(prefix string, id compact.NodeID, text string) func() {
 }
 
 // perfectInner renders the nodes of a perfect internal subtree.
-func perfectInner(prefix string, level uint, index uint64, top bool, nodeText nodeTextFunc) {
+func perfectInner(prefix string, level uint, index uint64, top bool, nodeText nodeTextFunc, dataText nodeTextFunc) {
 	id := compact.NewNodeID(level, index)
 	modifyNodeInfo(id, func(n *nodeInfo) {
-		n.leaf = id.Level == 0
 		n.perfectRoot = top
 	})
 
-	text := nodeText(id)
 	if level == 0 {
-		drawLeaf(prefix, index, text)
+		leafNodeText := nodeText(id)
```

Comment:
You could keep this line before `if level == 0 {`, and not duplicate it.

Reply:
I don't _think_ it's duplicated? We're passing text through here, and the funcs below - but your suggestion also indirectly suggests that this is suboptimal for grokking, so I've made `drawLeaf` take the funcs directly too.

Reply:
There were 2 branches, each calling `nodeText(id)` for the same ID. Now it looks better.